pkg/network/ndmz/dualstack.go

```diff
@@ -92,10 +92,16 @@ func (d *DualStack) Create() error {
 // Delete deletes the NDMZ network namespace
 func (d *DualStack) Delete() error {
 	netNS, err := namespace.GetByName(NetNSNDMZ)
-	if err == nil {
-		if err := namespace.Delete(netNS); err != nil {
-			return errors.Wrap(err, "failed to delete ndmz network namespace")
-		}
+	if err != nil {
```

Comment:
Here you risk to return an not exist error while you are trying to delete the ndmz.
That's why the code was put inside the `err == nil` block cause you only need to do something if the namespace actually exists. If it already doesn't exists, this method is a no-op.